lib/acpi/acpi.c

```diff
@@ -513,10 +615,11 @@ int acpi_device_type_get(ACPI_RESOURCE *res)
 	return type;
 }
 
-struct acpi_dev *acpi_device_get(char *hid, int inst)
+struct acpi_dev *acpi_device_get(char *hid, char *inst)
 {
 	struct acpi_dev *child_dev;
-	int i = 0, inst_id;
+	int i = 0;
+	char *inst_id;
```

Comment:
Maybe `uid_str` instead of `inst_id`? Also, since variables should be declared in the smallest possible scope where they're used, you should probably move this declaration inside the branches later in the function.